wgengine/magicsock/magicsock.go

```diff
@@ -884,23 +884,24 @@ func (c *Conn) determineEndpoints(ctx context.Context) ([]tailcfg.Endpoint, erro
 		c.setNetInfoHavePortMap()
 	}
 
-	if nr.GlobalV4 != "" {
-		addAddr(ipp(nr.GlobalV4), tailcfg.EndpointSTUN)
+	v4Addrs, v6Addrs := nr.GetGlobalAddrs()
+	for _, addr := range v4Addrs {
+		addAddr(addr, tailcfg.EndpointSTUN)
+	}
+	for _, addr := range v6Addrs {
+		addAddr(addr, tailcfg.EndpointSTUN)
+	}
 
+	if len(v4Addrs) == 1 {
```

Comment:
This is a subtle difference; previously, we'd always add the `tailcfg.EndpointSTUN4LocalPort` endpoint if `MappingVariesByDestIP` is true; now, we only do it if we get a single global address, but not two+. IMO we should probably keep the previous behaviour? Maybe something like `if len(v4Addrs) > 0` instead? Especially since there's little downside to advertising a single extra endpoint.

Reply:
The risk with excessive stun4localport is that on a NAT that attempts 1-1 mapping but has overlap/churn, it might see one particular user observe all incoming connection attempts for all peers behind the same NAT, which in a large office on an active network could become non-negligible, for example if one of those nodes in the office is an AD server or the like.

The selection of exactly 1 here alongside MappingVariesByIP means something specific: it means the NAT is actually full varies, not partial, as in something like a Palo Alto suffering from this issue.

Reply:
`>=1` is half-way between, I think this effectively retains the behavior you're looking for.